### PR TITLE
Feature/jc19 382 add searching

### PR DESCRIPTION
**Pull request by LocTran-saritasa**
Add searching

### CODE REVIEW

File: apps/vanilla/src/utils/resetLocalStorage.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { KEY_SEARCHING, KEY_SORTING } from '../constants';
+import { setValueToLocalStorage } from '../service/localStorage';
+
+/** Check for null or undefined element.*/
```
Comment (artem-sotnichenko-saritasa) [inline apps/vanilla/src/utils/resetLocalStorage.ts:4]: It is not checking for anything, I suppose you forget to change it.

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

File: apps/vanilla/src/constants/key.ts [hunk 2]
```diff
@@ -1,2 +1,3 @@
 export const KEY_ORDER = 'order';
 export const KEY_SORTING = 'sorting';
+export const KEY_SEARCHING = 'searching';
```
Comment (panov-aleksey-saritasa) [inline apps/vanilla/src/constants/key.ts:3]: I think you should clarify that these keys are related to local storage, either use docs or wrap it in namespace

File: apps/vanilla/src/pages/index.html [hunk 3]
```diff
@@ -9,6 +9,13 @@
   </head>
   <body class="default-layout">
     <h1 class="title">Saritasa Anime</h1>
+    <div class="search">
+      <div class="search__title">Search: </div>
```
Comment (panov-aleksey-saritasa) [inline apps/vanilla/src/pages/index.html:13]: Use [label tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for this

File: libs/core/models/paginationOptions.ts [hunk 4]
```diff
@@ -17,6 +17,9 @@ export class PaginationOptions extends Immerable {
   /** Total pages of data. */
   public readonly totalPages: number;
 
+  /** Search items by name. */
+  public readonly search: string;
+
```
Comment (panov-aleksey-saritasa) [inline libs/core/models/paginationOptions.ts:22]: Why do you mix up search string with pagination? Pagination and search string are unrelated things to each other

**Review by panov-aleksey-saritasa — CHANGES_REQUESTED** [review]

**Review by maxineo — APPROVED** [review]

File: apps/vanilla/src/global/base.css [hunk 5]
```diff
@@ -1,5 +1,5 @@
 html {
   font-size: 20px;
   font-family: system-ui, sans-serif;
-  color: var(--primary-font-color);
+  color: var(--font-color-primary);
```
Comment (chernodub) [inline apps/vanilla/src/global/base.css:4]: Seems like you don't have such a property in `apps/vanilla/src/global/variables.css`

File: apps/vanilla/src/pages/index.html [hunk 6]
```diff
@@ -9,6 +9,13 @@
   </head>
   <body class="default-layout">
     <h1 class="title">Saritasa Anime</h1>
+    <form class="form">
+      <label class="form__label">
+        <span class="form__label-text">Search:</span>
+        <input class="form__label-input" name="search" required />
```
Comment (chernodub) [inline apps/vanilla/src/pages/index.html:15]: Consider also specifying `type="search"` to improve the a11y and UX

File: apps/vanilla/src/scripts/renderSearching.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { DEFAULT_LIMIT, DEFAULT_SEARCH, KEY_SEARCHING } from '../constants';
+import { LocalStorageService } from '../service/localStorage';
+
+import { renderAnimeList } from './renderAnimeList';
+
+/**
+ * Render search result to UI.
+ * @param options Pagination option.
+ */
+export function searchingHandler(options: AnimeListQueryOptions): void {
```
Comment (chernodub) [inline apps/vanilla/src/scripts/renderSearching.ts:12]: [Functions should be named with a verb](https://wiki.saritasa.rocks/frontend/languages/ts-js/naming#naming-functions)

File: apps/vanilla/src/utils/resetLocalStorage.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { KEY_SEARCHING, KEY_SORTING } from '../constants';
+import { LocalStorageService } from '../service/localStorage';
+
+/** Reset value in local storage.*/
+export function resetStorage(): void {
+  LocalStorageService.setValue(KEY_SEARCHING, '');
```
Comment (chernodub) [inline apps/vanilla/src/utils/resetLocalStorage.ts:6]: Seems like this function does a very specific thing, it only clears sort and searching. But the name is too generic. From the name, I would assume that all the values are going to be cleared.

**Review by pavel-romanov8 — APPROVED** [review]

**Pull request closed without merging**